app/ts/Elicitation/ElicitationContext.tsx

```diff
@@ -16,7 +17,7 @@ export function ElicitationContextProviderComponent({
 }: {
   elicitationMethod: ElicitationMethod;
   cancel: () => void;
-  save: (preferences: IExactSwingRatio[]) => void;
+  save: (preferences: (IRatioBound | IExactSwingRatio)[]) => void;
```

Comment:
I think the typing should be 
IRatioBound[] | IExactSwingRatio[]

As it is now it allows for mixed preferences.